### PR TITLE
Fix IText rendering to take viewportTransform into account

### PR DESCRIPTION
**Pull request by fenderle**
When having a viewportTransform on the canvas IText fails to correctly render the cursor and the hidden area since it doesn't take the viewportTransform into account.

This patch fixes this, but I'm not entirely sure if I took the right approach.

### CODE REVIEW

File: src/shapes/itext.class.js [hunk 1]
```diff
@@ -497,7 +497,7 @@
       ctx.fillRect(
         boundaries.left + leftOffset,
         boundaries.top + boundaries.topOffset,
-        this.cursorWidth / this.scaleX,
+        this.cursorWidth / this.scaleX / this.canvas.viewportTransform[0],
```
Comment (asturur) [inline src/shapes/itext.class.js]: Before fillRect, calculate a scale factor that take in account scaleX and this.canvas.getZoom();
var scaleFactor = this.scaleX;
if (this.canvas) {
  scaleFactor *= this.canvas.getZoom();
}

and then use that scaleFactor to divide the cursorWidth.

File: src/mixins/itext_behavior.mixin.js [hunk 2]
```diff
@@ -430,8 +430,8 @@
       this.hiddenTextarea.selectionEnd = this.selectionEnd;
       if (this.selectionStart === this.selectionEnd) {
         var p = this._calcTextareaPosition();
```
Comment (asturur) [inline src/mixins/itext_behavior.mixin.js]: modify _calcTextareaPosition to take in account the viewporttransform.

before returning the point, use fabric.util.transformPoint(p, this.getViewportTransform());

so then we can use this modified function to take in account also the other case.

**Comment by asturur** [comment]
it misses translation but it was sometihng in the list of things to do. I think it misses also retina control.

**Comment by fenderle** [comment]
I applied your comments - hopefully I understood them correctly...

**Comment by asturur** [comment]
I'm getting a mac next week. waiting for it so i test this + retina screen and i will merge or ask you for tweak.

**Comment by asturur** [comment]
@fenderle i m working on a project where i need this to be fixed asap. Do you mind if i take over your work and finish it? i have to do it today.

**Pull request closed without merging**